src/components/FeaturePanel/PublicTransport/LineNumber.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+
+interface LineNumberProps {
+  name: string;
+  color: string;
+}
+
+/**
+ * A function to determine whether the text color should be black or white
+ * @param hexBgColor The background color in hex format, e.g. #ff0000
+ * @returns 'black' or 'white' depending on the brightness of the background color
+ */
+function whiteOrBlackText(hexBgColor) {
+  const r = parseInt(hexBgColor.slice(1, 3), 16);
+  const g = parseInt(hexBgColor.slice(3, 5), 16);
+  const b = parseInt(hexBgColor.slice(5, 7), 16);
+  const brightness = (r * 299 + g * 587 + b * 114) / 1000;
+  return brightness > 125 ? 'black' : 'white';
```

Comment:
very nice algorithm! 👍

Reply:
Works only when `colour` is hex. But it can be also W3C colour name: https://wiki.openstreetmap.org/wiki/Key:colour#How_to_map
Fails for this yellow line 8: https://osmapp.org/node/3325029085

Maybe implement mapping name -> hex for the _16 "basic colour names"_??